### PR TITLE
Move identity app to a library

### PR DESCRIPTION
**Pull request by hausdorff**
This PR is a first attempt to establish a consistent set of patterns for production stacks. At this point the most important thing to me is that we establish some agreement about what the API and README should look like, although of course other comments are also helpful (though perhaps less important).

### CODE REVIEW

File: aws/identity/lib/policies.ts [hunk 1]
```diff
@@ -1,28 +1,128 @@
...
+        args?: BaselineIamPoliciesOptions,
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super(`${common.groupName}:index:BaselineIamPolicies`, name, args, opts);
+
+        args = args === undefined ? {} : args;
+
+        //
+        // IAM Policies that we create by default.
+        //
```
Comment (metral) [inline aws/identity/lib/policies.ts]: this comment doesn't coincide with the code block right after it w.r.t. `args.defineUseExistingIamRolesPolicy`

File: aws/identity/lib/baseline.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    constructor(
+        name: string,
+        args: BaselineIamOptions = {},
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super(`${common.groupName}:index:BaselineIam`, name, args, opts);
+
+        const defineIamPassPolicy = args.policies && args.policies.defineUseExistingIamRolesPolicy;
+        let defineIamPassGroup = args.groups && args.groups.defineUseExistingIamRolesGroup;
```
[the remark this answers is not in the document]
Comment (hausdorff) [reply aws/identity/lib/baseline.ts]: I _always_ use `const`, but this time I felt like it was more readable to re-assign the variable than to have a gigantic nested tertiary expression. Hence, `let`.

File: aws/identity/lib/policies.ts (hunk 1 above)
Comment (hausdorff) [reply aws/identity/lib/policies.ts]: Hmm, I am pretty sure this is right, though perhaps confusing. Basically if `args.defineUseExistingIamRolesPolicy` is `undefined`, we want to set it to `true`; if not, just leave it how it is.

The effect of this is to default to `true`.

```
        args.defineUseExistingIamRolesPolicy =
            args.defineUseExistingIamRolesPolicy === undefined
                ? true
                : args.defineUseExistingIamRolesPolicy;
```

Comment (metral) [reply aws/identity/lib/policies.ts]: Makes sense. I'm still getting used to TS, and the ternary makes me think a bit

**Pull request merged**